src/permissions.ts

```diff
@@ -36,46 +53,17 @@ export class Permission {
   }
 }
 
-// deprecated
-export const AllowAll = new Permission('Allow all', MaxPriority, AnyResource, AnyRole, false, Allow);
-
-// deprecated
-export const DenyAll = new Permission('Deny all', MinPriority, AnyResource, AnyRole, false, Deny);
+type RequiredPolicyObject = Required<Policy>;
 
-export type Role = 'admin' | 'user' | 'guest' | 'anonymous' | 'channel_member' | 'channel_moderator' | string;
+export const Allow = 'Allow';
+export const Deny = 'Deny';
+export const AnyResource = ['*'];
+export const AnyRole = ['*'];
+export const MaxPriority = 999;
+export const MinPriority = 1;
 
-export const BuiltinRoles = {
-  Admin: 'admin',
-  Anonymous: 'anonymous',
-  ChannelMember: 'channel_member',
-  ChannelModerator: 'channel_moderator',
-  Guest: 'guest',
-  User: 'user',
-};
+// deprecated
```

Comment:
```suggestion
// @deprecated will be removed in the next major release
```
No idea when it would actually be removed, that's just guessing on my end 😅 But having it a jsdoc comment plays nice with most editors/LSP/whatever I think, does `// deprecated` do it normally as well? I might have to rewrite some config on my end for that :)